plugins/modules/ec2_instance.py

```diff
@@ -1630,6 +1631,45 @@ def value_wrapper(v):
     return changes_to_apply
 
 
+def change_instance_metadata_options(instance, params):
+    metadata_options_to_apply = params.get("metadata_options")
+
+    if metadata_options_to_apply is None:
+        return False
+
+    existing_metadata_options = camel_dict_to_snake_dict(instance.get("MetadataOptions"))
+
+    changes_to_apply = {
+        key: metadata_options_to_apply[key]
+        for key in set(existing_metadata_options) & set(metadata_options_to_apply)
+        if existing_metadata_options[key] != metadata_options_to_apply[key]
+    }
+
+    if not changes_to_apply:
+        return False
+
+    request_args = {
+        "InstanceId": instance["InstanceId"],
+        "HttpTokens": changes_to_apply.get("http_tokens", None),
+        "HttpPutResponseHopLimit": changes_to_apply.get("http_put_response_hop_limit", None),
+        "HttpEndpoint": changes_to_apply.get("http_endpoint", None),
+        "HttpProtocolIpv6": changes_to_apply.get("http_protocol_ipv6", None),
+        "InstanceMetadataTags": changes_to_apply.get("instance_metadata_tags", None),
+    }
```

Comment:
```suggestion
    request_args = {
        "InstanceId": instance["InstanceId"],
        "HttpTokens": changes_to_apply.get("http_tokens"),
        "HttpPutResponseHopLimit": changes_to_apply.get("http_put_response_hop_limit"),
        "HttpEndpoint": changes_to_apply.get("http_endpoint"),
        "HttpProtocolIpv6": changes_to_apply.get("http_protocol_ipv6"),
        "InstanceMetadataTags": changes_to_apply.get("instance_metadata_tags"),
    }
```

`.get()` defaults to returning `None`